File: C/Stack/Stack.cpp
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

typedef struct {
    int data[MAX_SIZE];
    int top; 
} Stack;

Stack* createStack() {
    Stack* stack = (Stack*)malloc(sizeof(Stack));
    stack->top = -1; 
    return stack;
}


int isFull(Stack* stack) {
    return stack->top == MAX_SIZE - 1;
}


int isEmpty(Stack* stack) {
    return stack->top == -1;
}


void push(Stack* stack, int value) {
    if (isFull(stack)) {
        printf("Stack dolu, eleman eklenemiyor.\n");
        return;
    }
    stack->data[++stack->top] = value; 
    printf("%d stack'e eklendi.\n", value);
}


int pop(Stack* stack) {
    if (isEmpty(stack)) {
        printf("Stack boş, eleman çıkarılamaz.\n");
        return -1;
    }
    return stack->data[stack->top--]; 
}


int peek(Stack* stack) {
    if (isEmpty(stack)) {
        printf("Stack boş, peek işlemi yapılamaz.\n");
        return -1;
    }
    return stack->data[stack->top];
}

int main() {
    Stack* stack = createStack(); 

    push(stack, 10); 
    push(stack, 20);
    push(stack, 30);

    printf("Stack'in üstündeki eleman: %d\n", peek(stack)); 

    printf("Eleman çıkarıldı: %d\n", pop(stack)); 
    printf("Eleman çıkarıldı: %d\n", pop(stack));

    printf("Stack'in üstündeki eleman: %d\n", peek(stack));

    push(stack, 40); 

    printf("Stack'in üstündeki eleman: %d\n", peek(stack));

    free(stack); 
    return 0;
}

File: Java/Stack/src/Stack.java
import java.util.EmptyStackException;

public class Stack {
    private int maxSize;
    private int[] stackArray;
    private int top;

    public Stack(int size) {
        this.maxSize = size;
        this.stackArray = new int[maxSize];
        this.top = -1;
    }

    public void push(int value) {
        if (isFull()) {
            System.out.println("Stack is full, cannot push " + value);
        } else {
            stackArray[++top] = value;
            System.out.println(value + " pushed to stack");
        }
    }

    public int pop() {
        if (isEmpty()) {
            throw new EmptyStackException();
        } else {
            int value = stackArray[top--];
            System.out.println(value + " popped from stack");
            return value;
        }
    }

    public int peek() {
        if (isEmpty()) {
            throw new EmptyStackException();
        } else {
            return stackArray[top];
        }
    }

    public boolean isEmpty() {
        return (top == -1);
    }

    public boolean isFull() {
        return (top == maxSize - 1);
    }

    public void display() {
        if (isEmpty()) {
            System.out.println("Stack is empty");
        } else {
            System.out.print("Stack (top to bottom): ");
            for (int i = top; i >= 0; i--) {
                System.out.print(stackArray[i] + " ");
            }
            System.out.println();
        }
    }

}